app/src/main/kotlin/com/camerasecuritysystem/client/SettingsActivity.kt

```diff
@@ -1,39 +1,108 @@
 package com.camerasecuritysystem.client
 
+import android.app.Activity
 import android.content.Context
 import android.content.SharedPreferences
 import android.os.Bundle
+import android.util.Log
 import android.view.View
-import androidx.appcompat.app.AppCompatActivity
 import android.widget.Button
-import com.camerasecuritysystem.client.databinding.SettingsActivityBinding
+import androidx.appcompat.app.AppCompatActivity
+import com.camerasecuritysystem.client.databinding.ActivitySettingsBinding
+import com.camerasecuritysystem.client.models.ServerConnection
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.launch
 
 class SettingsActivity : AppCompatActivity(),
     ConnectDialog.ConnectDialogListener {
 
-    private var binding: SettingsActivityBinding? = null
+    private var binding: ActivitySettingsBinding? = null
 
     private var keyStore = KeyStoreHelper("connectToServer")
 
+    lateinit var connectionLiveData: ConnectionLiveData
+
+    lateinit var connectBtn : Button
+
     private lateinit var sharedPreferences: SharedPreferences
     var pwdIV: String? = null
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        setContentView(R.layout.settings_activity)
+        setContentView(R.layout.activity_settings)
 
         supportActionBar?.setDisplayHomeAsUpEnabled(true)
-        binding = SettingsActivityBinding.inflate(layoutInflater)
+        binding = ActivitySettingsBinding.inflate(layoutInflater)
 
         sharedPreferences =
             this.getSharedPreferences("com.camerasecuritysystem.client", Context.MODE_PRIVATE)
 
-        pwdIV = sharedPreferences.getString("pwdIVByte", "")
+        pwdIV = sharedPreferences.getString(resources.getString(R.string.pwdIVByte), "")
 
-        val button = findViewById<View>(R.id.connectBtn) as Button
-        button.setOnClickListener {
+
+        val settings = findViewById<View>(R.id.textViewConnectionSettings)
+        settings.setOnClickListener {
             openDialog()
         }
+
+        connectBtn = findViewById<View>(R.id.connectBtn) as Button
+
+        connectBtn.setOnClickListener {
+
+            val context = this.applicationContext
+
+            //Try to setup a connection
+            GlobalScope.launch {
+                try {
+                    val connection = ServerConnection.getInstance()
+                    if (!connection.isConnected()){
+                        connection.connectIfPossible(context)
+
+                        Log.e("Connection", "Verbinden mogelijk: ${connection.connectIfPossible(context)}")
+                    }else{
+                        Log.e("Connection", "Ik ben al verbonden")
```

Comment:
We use English as our main language
```suggestion
                        Log.e("Connection", "Connecting is possible: ${connection.connectIfPossible(context)}")
                    }else{
                        Log.e("Connection", "Already connected")
```

Reply:
Or just remove logs?